### ISSUE TITLE
Hyperspec glossary

### ISSUE THREAD

**Issue by xiongtx** [slime/slime#288]
In `hyperspec.el`, there's a comment noting that the glossary functionality is not used. What is the current plan regarding this functionality, if any exists? Has it been superseded by something else, or is it just waiting for a pull request to make it functional?

**Comment by luismbo** [comment]
I'm not aware of any such plan. What's your idea?

**Comment by xiongtx** [comment]
Well, it would probably not be difficult to create a list of glossary terms and corresponding page anchors such that we can have a command like `common-lisp-hyperspec-glossary-term` that allows you to enter a glossary term and takes you to the corresponding Hyperspec section. Would that be useful? 

Just for some "cultural" context, is the Lispworks Hyperspec *the* documentation source for Common Lisp? It seems like it, given how old `hyperspec.el` is. If I understand correctly, the ANSI standard needs to be purchased, and the Hyperspec is non-free. I also recall that the non-free culture around Common Lisp was a major factor that drove RMS to start the FSF long, long ago. 

Is that why there isn't an Info version of CL documentation that can be accessed within Emacs, instead of having to launch browser windows? CLiki mentions [dpans2texi](http://users-phys.au.dk/harder/dpans.html); there's also Info documentation for GCL that is quite extensive but does not seems to be equivalent to the standard.

**Comment by luismbo** [comment]
It might be useful, yes, particularly if it can do some sort of completion or fuzzy matching as you mentioned. The l1sp.org redirection service includes glossary items too, http://l1sp.org/cl/glossary/binding, FWIW.

Lispwork's Hyperspec seems to be the canonical copy of the Hyperspec; it's what l1sp.org points to. (It's mirrored at clhs.lisp.se. Not sure how Lispworks feels about that.) I sometimes use this one at work: http://franz.com/support/documentation/current/ansicl/ansicl.htm.

GCL's one, which I thought was generated with dpans2texi, is pretty neat but it has some issues with nodes that contain more than one Lisp definition IIRC. I'd love to have an info version that addresses those issues and that shipped with SLIME!

**Comment by xiongtx** [comment]
I've noticed that for the IssueX functionality (which also seems to be unused), there's a variable `common-lisp-hyperspec-issuex-table` that has the following docstring:

> The HyperSpec IssueX table file.  If you copy the HyperSpec to your local system, set this variable to the location of the Issue cross-references table which is usually "Map_IssX.txt" or "Issue-Cross-Refs.text".

Is there any document such as "Map_IssX.txt" for glossary terms? Should we use it if there is one?

Right now I'm doing the following to get all the glossary terms:

1. Get a list of glossary pages.
2. Download pages to temp files.
3. Scan the pages for glossary terms and page anchors to generate a list of
   term-anchor pairs that can be put into a hash table.
4. Hard-code the list of pairs in the source code to add entries to the
   glossary term hash table.
5. Comment out code for steps 1-3 because they're no longer necessary
   once we've got all the pairs we need.
